crates/bevy_ecs/src/query/fetch.rs

```diff
@@ -712,6 +722,18 @@ unsafe impl<'a> QueryData for FilteredEntityRef<'a> {
         item
     }
 
+    #[inline]
+    fn provide_extra_access(
+        state: &mut Self::State,
+        access: &mut Access<ComponentId>,
+        available_access: &Access<ComponentId>,
+    ) {
+        state.clone_from(available_access);
```

Comment:
Comment here please: this is pretty tricky to understand when skimming out of context.